bundles/org.openhab.binding.homeconnect/src/main/resources/assets/js/homeconnect.js

```diff
@@ -0,0 +1,218 @@
+                console.log(`key=${key}  value=${responseHeader[key]}`);
+                responseHeaderElement.append($(`<dt class="col-sm-4">${key}</dt>`));
+                responseHeaderElement.append($(`<dd class="col-sm-8 text-break">${responseHeader[key]}</dd>`));
+                responseHeaderElement.append($('<div class="w-100"></div>'));
+            });
+        }
+
+        requestHeaderElement.empty();
+        Object.keys(requestHeader).forEach(key => {
+            console.log(`key=${key}  value=${requestHeader[key]}`);
+            requestHeaderElement.append($(`<dt class="col-sm-4">${key}</dt>`));
+            requestHeaderElement.append($(`<dd class="col-sm-8 text-break">${requestHeader[key]}</dd>`));
+            requestHeaderElement.append($('<div class="w-100"></div>'));
+        });
+
+        modal.modal('handleUpdate');
+    })
+
+    $('.reload-page').click(function () {
+        location.reload();
+    });
+
+    $('.request-chart').each(function(index, element) {
+        var bridgeId = $(this).data('bridge-id');
+        var chartElement = element;
+
+        function makeplot(bridgeId, chartElement) {
+            Plotly.d3.csv('requests?bridgeId=' + bridgeId + '&action=request-csv', function (data) {
+                processData(data, chartElement)
+            });
+        }
+
+        function processData(allRows, chartElement) {
+            console.log(allRows);
+            var x = [], y = [], standardDeviation = [];
+
+            for (var i = 0; i < allRows.length; i++) {
+                var row = allRows[i];
+                x.push( row['time'] );
+                y.push( row['requests'] );
+            }
+            console.log( 'X',x, 'Y',y, 'SD',standardDeviation );
+            makePlotly( x, y, standardDeviation, chartElement );
+        }
+
+        function makePlotly( x, y, standard_deviation, chartElement ){
+            var traces = [{
+                x: x,
+                y: y,
+                type: 'histogram',
+                histfunc: 'sum',
+                xbins: {
+                    size: 1000
+                }
+            }];
+
+            Plotly.newPlot(chartElement, traces,
+                           {
+                               xaxis: {
+                                   rangemode: 'nonnegative',
+                                   autorange: true,
+                                   title: '',
+                                   type: 'date'
+                               },
+                               yaxis: {
+                                   title: 'requests',
+                                   rangemode: 'nonnegative'
+                               }
+                           },
+                           {
+                               displayModeBar: false,
+                               responsive: true
+                           });
+        };
+
+        makeplot(bridgeId, chartElement);
+    });
+}())
```

Comment:
1:12 - 'Chart' is defined but never used. (no-unused-vars)
25:27 - There should be no space after this paren. (space-in-parens)
25:91 - Missing space before function parentheses. (space-before-function-paren)
26:57 - A space is required after ','. (comma-spacing)
26:62 - A space is required after ','. (comma-spacing)
28:28 - Missing space before function parentheses. (space-before-function-paren)
29:57 - A space is required after ','. (comma-spacing)
29:62 - A space is required after ','. (comma-spacing)
31:30 - Missing space before function parentheses. (space-before-function-paren)
55:30 - Missing space before function parentheses. (space-before-function-paren)
57:32 - There should be no space after this paren. (space-in-parens)
58:1 - Expected indentation of 16 spaces but found 32. (indent)
58:58 - Missing space before function parentheses. (space-before-function-paren)
59:1 - Expected indentation of 20 spaces but found 16. (indent)
59:61 - A space is required after ','. (comma-spacing)
59:66 - A space is required after ','. (comma-spacing)
60:1 - Expected indentation of 20 spaces but found 16. (indent)
61:1 - Expected indentation of 16 spaces but found 12. (indent)
62:32 - Missing space before function parentheses. (space-before-function-paren)
63:61 - A space is required after ','. (comma-spacing)
63:66 - A space is required after ','. (comma-spacing)
66:34 - Missing space before function parentheses. (space-before-function-paren)
90:30 - Missing space before function parentheses. (space-before-function-paren)
92:32 - There should be no space after this paren. (space-in-parens)
92:118 - Missing space before function parentheses. (space-before-function-paren)
93:1 - Expected indentation of 16 spaces but found 20. (indent)
93:65 - A space is required after ','. (comma-spacing)
93:70 - A space is required after ','. (comma-spacing)
94:1 - Expected indentation of 16 spaces but found 20. (indent)
95:1 - Expected indentation of 12 spaces but found 16. (indent)
96:32 - Missing space before function parentheses. (space-before-function-paren)
97:61 - A space is required after ','. (comma-spacing)
97:66 - A space is required after ','. (comma-spacing)
100:34 - Missing space before function parentheses. (space-before-function-paren)
163:38 - Missing space before function parentheses. (space-before-function-paren)
167:26 - Missing space before function parentheses. (space-before-function-paren)
173:29 - Missing space before function parentheses. (space-before-function-paren)
179:24 - There should be no space after this paren. (space-in-parens)
179:36 - There should be no space before this paren. (space-in-parens)
180:24 - There should be no space after this paren. (space-in-parens)
180:40 - There should be no space before this paren. (space-in-parens)
182:25 - There should be no space after this paren. (space-in-parens)
182:29 - A space is required after ','. (comma-spacing)
182:36 - A space is required after ','. (comma-spacing)
182:44 - A space is required after ','. (comma-spacing)
182:62 - There should be no space before this paren. (space-in-parens)
183:24 - There should be no space after this paren. (space-in-parens)
183:62 - There should be no space before this paren. (space-in-parens)
186:28 - Missing space before function parentheses. (space-before-function-paren)
186:29 - There should be no space after this paren. (space-in-parens)
186:68 - There should be no space before this paren. (space-in-parens)
198:1 - Expected indentation of 16 spaces but found 27. (indent)
199:1 - Expected indentation of 20 spaces but found 31. (indent)
200:1 - Expected indentation of 24 spaces but found 35. (indent)
201:1 - Expected indentation of 24 spaces but found 35. (indent)
202:1 - Expected indentation of 24 spaces but found 35. (indent)
203:1 - Expected indentation of 24 spaces but found 35. (indent)
204:1 - Expected indentation of 20 spaces but found 31. (indent)
205:1 - Expected indentation of 20 spaces but found 31. (indent)
206:1 - Expected indentation of 24 spaces but found 35. (indent)
207:1 - Expected indentation of 24 spaces but found 35. (indent)
208:1 - Expected indentation of 20 spaces but found 31. (indent)
209:1 - Expected indentation of 16 spaces but found 27. (indent)
210:1 - Expected indentation of 16 spaces but found 27. (indent)
211:1 - Expected indentation of 20 spaces but found 31. (indent)
212:1 - Expected indentation of 20 spaces but found 31. (indent)
213:1 - Expected indentation of 16 spaces but found 27. (indent)
214:10 - Unnecessary semicolon. (no-extra-semi)

Reply:
Fixed code
```js
/* globals Chart:false, feather:false, Plotly:false, requests:false */

(function () {
    'use strict'

    feather.replace();

    $(".redirectUri").text(window.location.href.substring(0, window.location.href.lastIndexOf('/homeconnect') + 12));

    $('#apiDetailModal').on('show.bs.modal', function (event) {
        var button = $(event.relatedTarget);
        var thingId = button.data('thing-id');
        var action = button.data('api-action');
        var titleText = button.data('title');
        var modal = $(this);
        var title = modal.find('.modal-title');
        var subTitle = modal.find('.modal-subtitle');
        var responseBodyElement = modal.find('.modal-response-body');

        responseBodyElement.text('Loading...');
        title.text(titleText);
        subTitle.text(thingId);
        modal.modal('handleUpdate');

        let jqxhr = $.get('appliances?thingId=' + thingId + '&action=' + action, function (data) {
            responseBodyElement.text(JSON.stringify(data, null, '\t'));
        });
        jqxhr.fail(function (data) {
            responseBodyElement.text(JSON.stringify(data, null, '\t'));
        })
        jqxhr.always(function () {
            modal.modal('handleUpdate');
        });
    })

    $('#rawCommandDetailModal').on('show.bs.modal', function (event) {
        var button = $(event.relatedTarget);
        var thingId = button.data('thing-id');
        var haId = button.data('ha-id');

        var modal = $(this);
        var subTitle = modal.find('.modal-subtitle');
        var inputPath = modal.find('#raw-path');
        var inputBody = modal.find('#raw-request-body');
        var submit = modal.find('#raw-submit');
        var responseBodyElement = modal.find('.modal-response-body');
        var responseTitle = modal.find('.raw-response-header');

        subTitle.text(thingId);
        responseBodyElement.text('');
        responseTitle.hide();
        inputPath.val('/api/homeappliances/' + haId + '/programs/active')
        modal.modal('handleUpdate');

        submit.click(function () {
            responseBodyElement.text('Loading...');
            let jqxhr = $.post('appliances?thingId=' + thingId + '&action=put-raw&path=' + inputPath.val(),
                inputBody.val(), function (data) {
                    responseBodyElement.text(JSON.stringify(data, null, '\t'));
                    responseTitle.show();
                });
            jqxhr.fail(function (data) {
                responseBodyElement.text(JSON.stringify(data, null, '\t'));
                responseTitle.show();
            })
            jqxhr.always(function () {
                modal.modal('handleUpdate');
            });
        });
    })

    $('#rawGetDetailModal').on('show.bs.modal', function (event) {
        var button = $(event.relatedTarget);
        var thingId = button.data('thing-id');
        var haId = button.data('ha-id');

        var modal = $(this);
        var subTitle = modal.find('.modal-subtitle');
        var inputPath = modal.find('#raw-get-path');
        var submit = modal.find('#raw-get-submit');
        var responseBodyElement = modal.find('.modal-response-body');
        var responseTitle = modal.find('.raw-response-header');

        subTitle.text(thingId);
        responseBodyElement.text('');
        responseTitle.hide();
        inputPath.val('/api/homeappliances/' + haId + '/programs')
        modal.modal('handleUpdate');

        submit.click(function () {
            responseBodyElement.text('Loading...');
            let jqxhr = $.post('appliances?thingId=' + thingId + '&action=get-raw&path=' + inputPath.val(), function (data) {
                responseBodyElement.text(JSON.stringify(data, null, '\t'));
                responseTitle.show();
            });
            jqxhr.fail(function (data) {
                responseBodyElement.text(JSON.stringify(data, null, '\t'));
                responseTitle.show();
            })
            jqxhr.always(function () {
                modal.modal('handleUpdate');
            });
        });
    })

    $('#requestDetailModal').on('show.bs.modal', function (event) {
        var button = $(event.relatedTarget);
        var requestId = button.data('request-id');
        var request = requests.find(item => item.id == requestId);
        var requestHeader = request.homeConnectRequest.header;
        var requestBody = request.homeConnectRequest.body;
        var modal = $(this);
        var requestBodyElement = modal.find('.modal-request-body');
        var title = modal.find('.modal-title');
        var responseBodyElement = modal.find('.modal-response-body');
        var requestHeaderElement = modal.find('.modal-request-header');
        var responseHeaderElement = modal.find('.modal-response-header');

        title.text(request.homeConnectRequest.method + ' ' + request.homeConnectRequest.url);

        if (requestBody) {
            requestBodyElement.text(requestBody);
            requestBodyElement.removeClass('text-muted')
        } else {
            requestBodyElement.text('Empty request body');
            requestBodyElement.addClass('text-muted')
        }

        if (request.homeConnectResponse && request.homeConnectResponse.body) {
            responseBodyElement.text(request.homeConnectResponse.body);
            responseBodyElement.removeClass('text-muted')
        } else {
            responseBodyElement.text('Empty response body');
            responseBodyElement.addClass('text-muted')
        }

        responseHeaderElement.empty();
        if (request.homeConnectResponse && request.homeConnectResponse.header) {
            var responseHeader = request.homeConnectResponse.header;
            Object.keys(responseHeader).forEach(key => {
                console.log(`key=${key}  value=${responseHeader[key]}`);
                responseHeaderElement.append($(`<dt class="col-sm-4">${key}</dt>`));
                responseHeaderElement.append($(`<dd class="col-sm-8 text-break">${responseHeader[key]}</dd>`));
                responseHeaderElement.append($('<div class="w-100"></div>'));
            });
        }

        requestHeaderElement.empty();
        Object.keys(requestHeader).forEach(key => {
            console.log(`key=${key}  value=${requestHeader[key]}`);
            requestHeaderElement.append($(`<dt class="col-sm-4">${key}</dt>`));
            requestHeaderElement.append($(`<dd class="col-sm-8 text-break">${requestHeader[key]}</dd>`));
            requestHeaderElement.append($('<div class="w-100"></div>'));
        });

        modal.modal('handleUpdate');
    })

    $('.reload-page').click(function () {
        location.reload();
    });

    $('.request-chart').each(function (index, element) {
        var bridgeId = $(this).data('bridge-id');
        var chartElement = element;

        function makeplot (bridgeId, chartElement) {
            Plotly.d3.csv('requests?bridgeId=' + bridgeId + '&action=request-csv', function (data) {
                processData(data, chartElement)
            });
        }

        function processData (allRows, chartElement) {
            console.log(allRows);
            var x = [], y = [], standardDeviation = [];

            for (var i = 0; i < allRows.length; i++) {
                var row = allRows[i];
                x.push(row['time']);
                y.push(row['requests']);
            }
            console.log('X', x, 'Y', y, 'SD', standardDeviation);
            makePlotly(x, y, standardDeviation, chartElement);
        }

        function makePlotly (x, y, standard_deviation, chartElement){
            var traces = [{
                x: x,
                y: y,
                type: 'histogram',
                histfunc: 'sum',
                xbins: {
                    size: 1000
                }
            }];

            Plotly.newPlot(chartElement, traces,
                {
                    xaxis: {
                        rangemode: 'nonnegative',
                        autorange: true,
                        title: '',
                        type: 'date'
                    },
                    yaxis: {
                        title: 'requests',
                        rangemode: 'nonnegative'
                    }
                },
                {
                    displayModeBar: false,
                    responsive: true
                });
        }

        makeplot(bridgeId, chartElement);
    });
}())
```

Reply:
@ghys Thanks!

Reply:
Thank you